src/ILCompiler.Compiler/src/IL/Stubs/StartupCode/NativeLibraryStartupMethod.cs

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+using Internal.TypeSystem;
+
+using AssemblyName = System.Reflection.AssemblyName;
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.IL.Stubs.StartupCode
+{
+    /// <summary>
+    /// Startup code that does initialization, Main invocation
+    /// and shutdown of the runtime.
+    /// </summary>
+    public sealed partial class NativeLibraryStartupMethod : ILStubMethod
+    {
+        private TypeDesc _owningType;
+        private MethodSignature _signature;
+        private IList<MethodDesc> _libraryInitializers;
+
+        public NativeLibraryStartupMethod(TypeDesc owningType, IList<MethodDesc> libraryInitializers)
+        {
+            _owningType = owningType;
+            _libraryInitializers = libraryInitializers;
+        }
+
+        protected override int ClassCode => -304225482;
+
+        protected override int CompareToImpl(MethodDesc other, TypeSystemComparer comparer)
+        {
+            // Should be a singleton
+            Debug.Assert(this == other);
+            return 0;
+        }
+
+        public override TypeSystemContext Context
+        {
+            get
+            {
+                return _owningType.Context;
+            }
+        }
+
+        public override TypeDesc OwningType
+        {
+            get
+            {
+                return _owningType;
+            }
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return "NativeLibraryStartup";
+            }
+        }
+
+        public override MethodIL EmitIL()
+        {
+            ILEmitter emitter = new ILEmitter();
+            ILCodeStream codeStream = emitter.NewCodeStream();
+
+            // Allow the class library to run explicitly ordered class constructors first thing in start-up.
+            if (_libraryInitializers != null)
+            {
+                foreach (MethodDesc method in _libraryInitializers)
+                {
+                    codeStream.Emit(ILOpcode.call, emitter.NewToken(method));
+                }
+            }
+
+            // Simply return 0
```

Comment:
We could also just make this return `void`.

Reply:
I return `0` to signify a success exit code. That way this line: https://github.com/tonerdo/readline/blob/6d247be638fb9b1005c0b002cfb0f3c2bebced41/src/ReadLine/KeyHandler.cs#L276 will return the necessary value for the main function to continue

Reply:
If this can never return failure exit code, it is not very useful for it to return success exit code...

Also, it would be nice to change this method to take no arguments since they are unused and always null. It should make some of the ifdefs in the boostrapper simpler.